source/common/stream_info/stream_info_impl.h

```diff
@@ -332,6 +338,7 @@ struct StreamInfoImpl : public StreamInfo {
   std::string route_name_;
   absl::optional<uint64_t> upstream_connection_id_;
   absl::optional<uint32_t> attempt_count_;
+  std::string virtual_cluster_name_;
```

Comment:
per above let's try to avoid deep-copy here.

Reply:
Done.